flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/TestingCheckpointIDCounter.java

```diff
@@ -91,11 +92,14 @@ public static Builder builder() {
     /** {@code Builder} for creating {@code TestingCheckpointIDCounter} instances. */
     public static class Builder {
 
-        private Runnable startRunnable;
-        private Function<JobStatus, CompletableFuture<Void>> shutdownFunction;
-        private Supplier<Integer> getAndIncrementSupplier;
-        private Supplier<Integer> getSupplier;
-        private Consumer<Long> setCountConsumer;
+        private final AtomicLong defaultCheckpointIdStore = new AtomicLong();
```

Comment:
Wouldn't it be safer to use a local variable or a parameter in the `build()` method rather than a field? The current approach makes the `AtomicLong` been shared between different `TestingCheckpointIDCounter` instances that are created by the same builder. :thinking:

Reply:
Indeed, this needs to be fixed.